File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/BlockBreakAnimationPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record BlockBreakAnimationPacket(int entityId, int x, int y, int z, byte destroyStage)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        writeVarInt(os, entityId);
        os.writeInt(x);
        os.writeInt(y);
        os.writeInt(z);
        os.writeByte(destroyStage);
    }

    @Override
    public byte getId() {
        return 0x25;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/BlockBreakAnimationPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class BlockBreakAnimationPacket(val entityId: Int, val x: Int, val y: Int, val z: Int, val destroyStage: Byte) :
    Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        writeVarInt(os, entityId)
        os.writeInt(x)
        os.writeInt(y)
        os.writeInt(z)
        os.writeByte(destroyStage.toInt())
    }

    override fun getId(): Byte {
        return 0x25
    }
}
